docs/advanced_topics.rst

```diff
@@ -1,114 +1,251 @@
-Advanced topics
+Advanced Topics
 =================
 
-This section covers a few advanced topics that are mentioned in the API description. However, for the typical use case the default DALI configuration will perform well out of the box, and you don't need to worry about these advanced topics.
+This section covers a few advanced topics that are mentioned in the API documentation.
 
-DALI internal data format
+.. note::
+  For typical use cases, the default DALI configuration performs well out of the box, and you do
+  not need to review this section.
+
+DALI Internal Data Format
 -------------------------
 
-Internally, DALI processes data in NHWC format. To reshape it to NCHW, please use ``\*Permute`` operator family. Also note that if such operators are used in the middle of a pipeline, then the subsequent operations in the pipeline will make incorrect assumptions about the processed data format, and the result will be undefined.
+Internally, DALI processes data in the [N, Width, Height, Channels] (HNHWC) format. To reshape it
+to NCHW, use the ``\*Permute`` operator family. If these operators are used in the middle of
+a pipeline, the subsequent operations in the pipeline will make incorrect assumptions about the
+processed data format, and the result will be undefined.
 
-Thread affinity
+Thread Affinity
 ---------------
 
-The purpose of this functionality is to allow pinning DALI threads to the given CPU. Thread affinity can save the overhead of worker threads jumping from core to core. Hence this functionality can increase the performance for CPU heavy workloads.
-It is possible to set the CPU thread affinity for DALI using the ```DALI_AFFINITY_MASK``` environment variable. ```DALI_AFFINITY_MASK`` should be a comma separated list of CPU ids numbers which will be assigned to corresponding DALI threads.
-The number of DALI threads are set during the pipeline construction by num_threads argument, and the set_affinity enables setting the thread affinity for the CPU worker threads. Note that for performance reasons the hybrid ```ImageDecoder``` operator (nvJPEG based) houses threads on its own, and they are always affined.
-If the number of threads are more than the CPU id numbers in the environment variable ``DALI_AFFINITY_MASK``, then the following applies: First, the threads are assigned to CPU id numbers until all CPU id numbers from ``DALI_AFFINITY_MASK are assigned``. Then, for the remaining threads, the CPU id numbers from nvmlDeviceGetCpuAffinity will be used. See below example:
+This functionality allows you to pin DALI threads to the specified CPU. Thread affinity saves
+the overhead of worker threads jumping from core to core and improves the CPU heavy workload performance.
+You can set the DALI CPU thread affinity by using the ``DALI_AFFINITY_MASK`` environment variable,
+which is a comma-separated list of CPU IDs that will be assigned to corresponding DALI threads.
```

Comment:
```suggestion
DALI CPU thread affinity is controlled by the ``DALI_AFFINITY_MASK`` environment variable,
which contains a comma-separated list of CPU IDs that will be assigned to consecutive DALI threads.
```
There's no DALI thread identifier for correspondence - if they are assigned sequentially, then "consecutive" is a better word.